### DISCUSSION TITLE
typing.Callable and type hint of decorated method

### DISCUSSION THREAD

**Discussion by paivan1999** [coady/multimethod#99] [Q&A] [answered]
Hello from Russia,
I've tried your multimethod library (version 1.10) recently, and 've found some problems.
1. typing.Callable is still not supported (I've seen that as you said, you've fixed it in 1.8, but problem remains in newer version)

my code:
class A:
    @multimethod
    def m(self, c: Callable[[int],int]) -> None:
        print("helloCallable")

if __name__ == '__main__':
    t:Callable[[int],int] = lambda arg:arg
    A().m(t)

error code:
Traceback (most recent call last):
  File "....py", line 33, in <module>
    A().m(t)
  File "/home/.../.local/lib/python3.8/site-packages/multimethod/__init__.py", line 326, in __call__
    func = self[tuple(func(arg) for func, arg in zip(self.type_checkers, args))]
  File "/home/.../.local/lib/python3.8/site-packages/multimethod/__init__.py", line 320, in __missing__
    raise DispatchError(msg, types, keys)
multimethod.DispatchError: ('m: 0 methods found', (<class '__main__.A'>, <class 'collections.abc.Callable'>), [])

2.
i'm using PyCharm, and find it more comfort to see all possible types of overloaded function when type it, like in multipledispatch, but in multimethod it just shows only one type with *args,**kwargs signature, while there are more than one overload with precise signature. Picture is added here:
![image](https://github.com/coady/multimethod/assets/81824818/23eb90ab-223e-4317-8623-04bef4a81ccd)

I'll be grateful to see fixes of this problems

**Comment by paivan1999** [comment]
p.s.
If not these problems, i would definitely prefer multimethod to multipledispatch

**Comment by coady** [comment, marked as answer]
Your example works if `t` is defined as:
```python
def t(arg: int) -> int:
        return arg
```
Whereas with `t:Callable[[int],int] = lambda arg:arg`, there is no way to know the function signature.

**Reply by paivan1999** [reply]
yup, I was not right at this point. Thank you.
But what with multimethod signature prompt? is multipledispatch better in this or I can fix it by myself?
